File: cpp/remove_element.cpp
class Solution {
public:
    int removeElement(int A[], int n, int elem) {
        int length = 0;
        for (int i=0; i<n; ++i)
        {
            if (A[i] != elem)
            {
                A[length] = A[i]; 
                length++;
            }
        }
        return length;
    }
};

File: python/remove_element.py
class Solution:
    # @param    A       a list of integers
    # @param    elem    an integer, value need to be removed
    # @return an integer
    def removeElement(self, A, elem):
        length = 0
        for i in range(len(A)):
            if A[i] != elem:
                A[length] = A[i]
                length += 1
        return length